x-pack/plugins/observability_solution/entities_data_access/server/services/index_pattern_service.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObjectsClientContract } from '@kbn/core-saved-objects-api-server';
+import { EntityDefinition } from '@kbn/entities-schema';
+import {
+  ENTITY_BASE_PREFIX,
+  ENTITY_HISTORY,
+  ENTITY_LATEST,
+  SO_ENTITY_DEFINITION_TYPE,
+} from '../../common';
+import { RegisterServicesParams } from '../types';
+
+type SchemaVersionString = `v${number}`;
+
+interface Options {
+  datasets?: 'all' | typeof ENTITY_LATEST | typeof ENTITY_HISTORY;
+  schemaVersion?: SchemaVersionString | SchemaVersionString[] | 'all';
+}
+
+function translateDatasetsOption(datasets: Options['datasets']) {
+  if (!datasets || datasets === 'all') {
+    return ['*'] as const;
+  }
+  return [datasets] as const;
+}
+
+function translateSchemaVersionOption(schemaVersion: Options['schemaVersion']) {
+  if (!schemaVersion || schemaVersion === 'all') {
+    return ['*'] as const;
+  }
+
+  return Array.isArray(schemaVersion) ? schemaVersion : [schemaVersion];
+}
+
+async function findAllDefinitionsByType(type: string, soClient: SavedObjectsClientContract) {
```

Comment:
Can we type the `type` argument here? instead of using `string`?

Reply:
Thing is that it can technically be any string as the type, in order to support custom types added by users!